### PR TITLE
Add Heroku Add-on

### PR DESCRIPTION
**Pull request by noor-ahmed**
## What happened
Solved https://github.com/nimblehq/rails-templates/issues/218
 
## Insight

- [x] Added add-on option to `template.rb`

If Dev wants to add Heroku, then
- [x] Added Heroku Configuration to `production.rb`
- [x]  Added variables to `.env`

## Proof Of Work
Screen Shot 2020-10-09 at 7 03 24 PM

### CODE REVIEW

File: .template/addons/heroku/config_template.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+insert_into_file 'config/environments/production.rb', before: /%r{end\nend\Z}/ do
+  <<~CODE
+    config.public_file_server.enabled = ENV['RAILS_SERVE_STATIC_FILES'].present?
```
Comment (gutakk) [inline .template/addons/heroku/config_template.rb:3]: I think we don't need this line. As we had [this](https://github.com/nimblehq/rails-templates/pull/242)

File: .template/addons/heroku/env_variables.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+echo 'RAILS_SERVE_STATIC_FILES=true' >> '.env' # default value
```
Comment (gutakk) [inline .template/addons/heroku/env_variables.rb:1]: Same as this [comment](https://github.com/nimblehq/rails-templates/pull/238#discussion_r503026561)

File: template.rb [hunk 3]
```diff
@@ -60,6 +60,12 @@ def apply_template!(template_root)
   apply '.template/addons/docker/template.rb'
   apply '.template/addons/semaphore/template.rb'
 
+  if ENV['CI']
+    apply '.template/addons/heroku/template.rb'
+  elsif yes?("Would you like to add heroku?")
```
Comment (gutakk) [inline template.rb:65]: How about add an action to guide user?

something like this

```ruby
Would you like to add Heroku? (yes/no)
```

File: .template/addons/heroku/config_template.rb (hunk 1 above)
Comment (noor-ahmed) [reply .template/addons/heroku/config_template.rb]: @gutakk Thanks for the good catch, resolved https://github.com/nimblehq/rails-templates/pull/238/commits/fb49d58544d911d81c69b8da4041687dfa85a460 🤝

File: .template/addons/heroku/env_variables.rb (hunk 2 above)
Comment (noor-ahmed) [reply .template/addons/heroku/env_variables.rb]: resolved https://github.com/nimblehq/rails-templates/pull/238/commits/fb49d58544d911d81c69b8da4041687dfa85a460

File: template.rb (hunk 3 above)
Comment (noor-ahmed) [reply template.rb:65]: @gutakk yeah that I have also in my mind. We will make it an one line once reviewing done and ready for merging ✌️

**Comment by ankitkalia1195** [comment]
@andyduong1920 Should I close this PR, as it is opened since long time and @noor-ahmed no longer works at the company. Also in this PR, I dont see anything related to heroku 🤔 . The issue you opened also does not contain any description https://github.com/nimblehq/rails-templates/issues/218.

**Comment by andyduong1920** [comment]
> @andyduong1920 Should I close this PR, as it is opened since long time and noor no longer works at the company. Also in this PR, I dont see anything related to heroku 🤔 . The issue you opened also does not contain any description #218.

Yeah sure @ankitkalia1195 👍

**Pull request closed without merging**